classifier_logging.py

```diff
@@ -0,0 +1,50 @@
+import logging
+import os 
+import pandas as pd
+import time
+
+logging.basicConfig(
+    format = '%(asctime)s - %(levelname)s - %(name)s -   %(message)s',
+    datefmt = '%m/%d/%Y %H:%M:%S',
+    level = logging.INFO
+)
+
+logger = logging.getLogger(__name__)
+
+
+def log_training(num_examples, train_batch_size, num_train_optimization_steps):
+    """ logs the beginning of training """
+    logger.info("***** Running training *****")
+    logger.info("  Num examples = %d", num_examples)
+    logger.info("  Batch size = %d", train_batch_size)
+    logger.info("  Num steps = %d", num_train_optimization_steps)
+
+def log_evaluating(num_eval_examples, eval_batch_size):
+    """ Logs the beginning of the evaluation. """ 
+    logger.info("***** Running evaluation *****")
+    logger.info("  Num examples = %d", num_eval_examples)
+    logger.info("  Batch size = %d", eval_batch_size)
+
+def log_result(output_dir, num_train_examples, num_test_examples, result):
+    """ logs and writes the evalutated stats of the model """
+    current_time = time.strftime("%m-%d_%I-%M%p")
+    output_eval_file = os.path.join(output_dir,"eval_results", "result-{}.txt".format(current_time))
+    with open(output_eval_file, "w") as writer:
+        writer.write("train_examples = %d\n" % (num_train_examples,))
+        writer.write("test_examples = %d\n\n" % (num_test_examples,))
+        logger.info("***** Eval results *****")
+        for key in sorted(result.keys()):
+            logger.info("  %s = %s", key, str(result[key]))
+            writer.write("%s = %s\n" % (key, str(result[key])))
+
+def record_to_csv(output_dir, num_train_examples, num_test_examples, result):
+    """ creates (if necessary) and stores results in a 'combined.csv' file """
+    path = os.path.join(output_dir, "combined.csv")
+    result["num_train_examples"] = num_train_examples
+    result["num_test_examples"] = num_test_examples
+    results = pd.Series(result)
+    past_results = pd.DataFrame()
```

Comment:
This part might be clearer as
```
if os.path.exists(path):
    past_results = pd.read_csv(path)
else:
    past_results = pd.DataFrame()
```

(just to make it super obvious that `past_results` always exists,  and therefore we will be able to append to it in line 49.)

What happens if `pd.read_csv` throws an exception? (maybe the path is a directory, maybe the file isn't a valid csv file, et cetera)? Do you want to let that bubble up and crash the program (this can be the right call if there's no obvious way for the computer to handle it, or if you just don't know what to expect so you want things to break dramatically if they break so that you will notice)? Do you want to write the current results to CSV and just ignore the past results? Do you want to not bother writing to CSV, display some kind of error message, and let the rest of the program keep executing? (There is no particular right answer here; just that the code should reflect what you pick.)